### PR TITLE
More NtOS interfaces, random fixes, client-side theming

### PR DESCRIPTION
**Pull request by stylemistake**
## About The Pull Request

- Random fixes
- Theme is now handled on the client-side, via `routes.js` file.
- Fixes a small Tab component regression.

### NtOS Net Downloader

ntdl

## Why It's Good For The Game

Setting themes on client side is good, because: a. you save bandwidth on ui_data; b. you can experiment with new themes without the need to set ui_style and recompile the whole server.

New UI good.

## Changelog
:cl:
add: New NtOS Net Downloader interface.
tweak: UIs will not be greyed out if you're a ghost, for better observing experience.
fix: Fix the sheet progress bar in Portable Generator.
/:cl:

<!--
add: Added more things
del: Removed old things
tweak: tweaked a few things
balance: rebalanced something
fix: fixed a few things
soundadd: added a new sound thingy
sounddel: removed an old sound thingy
imageadd: added some icons and images
imagedel: deleted some icons and images
spellcheck: fixed a few typos
code: changed some code
refactor: refactored some code
config: changed some config setting
admin: messed with admin stuff
server: something server ops should know

### CODE REVIEW

File: code/modules/modular_computers/file_system/programs/ntdownloader.dm [hunk 1]
```diff
@@ -113,49 +115,50 @@
...
+		if(!P.can_run(user,transfer = 1) || hard_drive.find_file_by_name(P.filename))
+			continue
+		all_entries.Add(list(list(
+		"filename" = P.filename,
+		"filedesc" = P.filedesc,
+		"fileinfo" = P.extended_desc,
+		"compatibility" = check_compatibility(P),
+		"size" = P.size
+		)))
+	data["hackedavailable"] = 0
```
Comment (actioninja) [inline code/modules/modular_computers/file_system/programs/ntdownloader.dm]: = FALSE

File: code/modules/modular_computers/file_system/programs/ntdownloader.dm [hunk 2]
```diff
@@ -113,49 +115,50 @@
...
+	if(computer.obj_flags & EMAGGED) // If we are running on emagged computer we have access to some "bonus" software
+		var/list/hacked_programs[0]
+		for(var/S in SSnetworks.station_network.available_antag_software)
+			var/datum/computer_file/program/P = S
+			if(hard_drive.find_file_by_name(P.filename))
 				continue
-			all_entries.Add(list(list(
+			data["hackedavailable"] = 1
```
Comment (actioninja) [inline code/modules/modular_computers/file_system/programs/ntdownloader.dm]: = TRUE

**Review by actioninja — CHANGES_REQUESTED** [review]

File: code/modules/modular_computers/file_system/programs/ntdownloader.dm [hunk 3]
```diff
@@ -113,49 +115,50 @@
 
 	var/list/data = get_header_data()
 
-	// This IF cuts on data transferred to client, so i guess it's worth it.
-	if(downloaderror) // Download errored. Wait until user resets the program.
-		data["error"] = downloaderror
-	else if(downloaded_file) // Download running. Wait please..
+	data["downloading"] = !!downloaded_file
```
[the remark this answers is not in the document]
Comment (stylemistake) [reply code/modules/modular_computers/file_system/programs/ntdownloader.dm]: downloaded_file is a string, and it does convert it into a "boolean", as in 0/1.

[the remark this answers is not in the document]
Comment (actioninja) [reply code/modules/modular_computers/file_system/programs/ntdownloader.dm]: eh, ok.

**Pull request merged**